utils/file-manager.js

```diff
@@ -288,46 +275,39 @@ export default class FileManager {
 
   /**
    * Return the file path(s) that match the ID.
-   * If multiple files are found and `unique` is set to true,
-   * print a warning and return only the first one.
-   * If no file is found, return null.
    * @param {String} id
    * @param {Boolean} unique return a single result
-   * @returns a String (unique=true), an array (unique=false), or null
+   * @returns a String / null (unique=true), or an array (unique=false)
    */
   getFilePathsFromId(id, unique = true) {
     if (!(id in this.idToGlob)) {
-      return null
+      throw new Error('ID is not defined')
     }
     const glob = this.idToGlob[id]
     const paths = this.findMatchingFilePaths(glob)
-    if (paths.length === 0) {
-      return null
-    } else if (paths.length > 1) {
-      if (unique) {
+    if (unique) {
+      if (paths.length === 0) {
+        return null
+      } else if (paths.length > 1) {
         console.warn(`Multiple files were found for id="${id}", glob="${glob}"`)
-      } else {
-        return paths
       }
+      return paths[0]
     }
-    return paths[0]
+    return paths
   }
 
   /**
    * Return the preprocessed text of the file(s) that match the ID.
-   * If multiple files are found and `unique` is set to true,
-   * print a warning and return only the first one.
-   * If no file is found, return null.
    * @param {String} id
    * @param {Boolean} unique return a single result
-   * @returns a String (unique=true), an array (unique=false), or null
+   * @returns a String / null (unique=true), or an array (unique=false)
    */
   async getPreprocessedTextFromId(id, unique = true) {
     const paths = this.getFilePathsFromId(id, unique)
-    if (paths === null) {
-      return null
-    }
     if (unique) {
+      if (paths === null) {
+        return null
+      }
       return await this.getPreprocessedText(paths)
```

Comment:
It's a bit confusing that the variable "paths" refers to just one path.

It works because getFilePathsFromId(id, true) returns a string, while getFilePathsFromId(id, false) returns an array.

I would like it better if these methods always returned the same type, an array. When unique=true, it would just return an array of length 1 at most.

Reply:
> That would make it easier to understand what they do

I agree

> would simplify the code where they're called.

This I don't agree, for instance in the facebook pipeline it becomes:
`(await fileManager.getPreprocessedTextFromId('advertisers-interacted')[0]) ?? null`
instead of
`await fileManager.getPreprocessedTextFromId('advertisers-interacted')`
for the same result (if there is no match we need null instead of undefined, otherwise JSONPath will crash)

> When unique=true, it would just return an array of length 1 at most.

If we always return an array, then `unique` is not really useful because the pipelines can ignore the rest of the array without additional code. We can remove it in this case

So in summary it's kind of a trade-off between the simplicity of the API and the conciseness in the usage. I'm happy to do as you suggest, but I just wanted to point out the differences

Reply:
Ok, I'm happy with my side of the trade-off, it simplifies methods from the library that will be used in many places, at the cost of giving the caller a little bit more work to make his code look good. I think he's capable of that.

JSONPath likes null better than undefined but in other places we'll prefer null. It's the caller's job to deal with things like these.